src/networking.c

```diff
@@ -3585,6 +3614,73 @@ static int parseClientFiltersOrReply(client *c, int index, clientFilter *filter)
                 return C_ERR;
             }
             index += 2;
+        } else if (!strcasecmp(c->argv[index]->ptr, "minidle") && moreargs) {
+            long long tmp;
+
+            if (getLongLongFromObjectOrReply(c, c->argv[index + 1], &tmp,
+                                             "minidle is not an integer or out of range") != C_OK)
+                return C_ERR;
+            if (tmp <= 0) {
+                addReplyError(c, "minidle should be greater than 0");
+                return C_ERR;
+            }
+
+            filter->min_idle = tmp;
+            index += 2;
+        } else if (!strcasecmp(c->argv[index]->ptr, "flags") && moreargs) {
+            filter->flags = c->argv[index + 1]->ptr;
+            index += 2;
+        } else if (!strcasecmp(c->argv[index]->ptr, "name") && moreargs) {
+            filter->name = c->argv[index + 1]->ptr;
+            index += 2;
+        } else if (!strcasecmp(c->argv[index]->ptr, "subscribed-pattern") && moreargs) {
+            filter->subscribed_pattern = createObject(OBJ_STRING, sdsnew(c->argv[index + 1]->ptr));
```

Comment:
Can't we just use the objects in argv without copying? No need to copy it and free it afterwards, or am I missing something?

```suggestion
            filter->subscribed_pattern = c->argv[index + 1];
```

Reply:
@zuiderkwast these field needs to be robj with sds type in order to be compared with client fields using `dictFind`. It was not working with just assigning the values like `c->argv[index + 1];`

This was the check: https://github.com/valkey-io/valkey/blob/unstable/src/server.c#L496

Reply:
OK, interesting. Makes sense. It's possible to do an optimization though: If it's already an sds-robj, we could increment the reference counter instead of duplicating the object.

Reply:
thanks, made this change as well!